php/src/vaas/Vaas.php

```diff
@@ -73,314 +69,321 @@ public function withLogger(LoggerInterface $logger): self
-                    $fileStream,
-                    $verdictResponse->url,
-                    $verdictResponse->upload_token,
-                    $fileSize,
-                    $verdictResponse->guid)
-            );
-
-            $fileStream->close();
-
-            return $verdict;
-        }
+            $stream = openFile($path, 'r');
 
-        return new VaasVerdict($verdictResponse);
+            $forStreamOptions = new ForStreamOptions($options->useHashLookup, 300, $options->vaasRequestId);
+
+            $this->logger->debug("Requesting verdict for $path as file stream");
+            return $this->forStreamAsync($stream, filesize($path), $forStreamOptions)->await();
+        });
     }
 
     /**
-     * Gets verdict by stream
-     *
-     * @param ReadableStream $stream   the path to get the verdict for
-     * @param bool   $upload should the file be uploaded if initial verdict is unknown
-     * @param string $uuid   unique identifier
-     * 
-     * @throws JsonMapper_Exception
-     * @throws VaasClientException
-     * @throws TimeoutException
-     * @throws VaasServerException
-     * @throws BadOpcodeException
-     * @throws VaasInvalidStateException
-     * @throws UploadFailedException
+     * Get the verdict for a stream
+     * @param ReadableStream $stream The stream to check
+     * @param int $fileSize The size of the file in bytes
+     * @param ForStreamOptions|null $options Options for the request
+     * @param Cancellation|null $cancellation Cancellation token
+     * @return Future A future that resolves to a VaasVerdict
      */
-    public function ForStream(ReadableStream $stream, int $size = 0, string $uuid = null): VaasVerdict
+    public function forStreamAsync(ReadableStream $stream, int $fileSize, ?ForStreamOptions $options = null, ?Cancellation $cancellation = null): Future
     {
-        $this->logger->debug("uuid: ".var_export($uuid, true));
-        $uuid = $uuid ?? UuidV4::getFactory()->uuid4()->toString();
-        $this->logger->debug("uuid: ".var_export($uuid, true));
-        $verdictResponse = $this->_verdictResponseForStream($uuid);
+        return async(function () use ($stream, $fileSize, $options, $cancellation) {
+            try {
+                $this->logger->debug("Requesting verdict for stream");
+
+                if (!$stream->isReadable() || $stream->isClosed()) {
+                    $this->logger->error("Stream is not readable");
+                    throw new VaasClientException('Stream is not readable');
+                }
+
+                $options = $options ?? ForStreamOptions::fromVaasOptions($this->options);
+                
+                $url = sprintf('%s/files?useHashLookup=%s', $this->options->vaasUrl, json_encode($options->useHashLookup));
+
+                $request = new Request($url, 'POST');
+
+                $request->setBody(StreamedContent::fromStream($stream, $fileSize));
+                $request->setTransferTimeout($options->timeout);
```

Comment:
This timeout along with its 300s default effectively means that users with slow connections can never upload medium to large files. Are we confident that this is the default behavior we want? Is there a significant harm to not time-outing uploads?